src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/AuditModel.java

```diff
@@ -0,0 +1,52 @@
+package com.amazon.opendistroforelasticsearch.security.securityconf.impl;
```

Comment:
Please clarify the object model. I don't see why it is necessary to introduce `AuditModel` interface with `AuditModelImpl` implementation that delegates everything to `Audit`.

Reply:
I have followed the pattern for InternalUsersModel, ConfigModel ..

The underlying implementation class Audit becomes version specific. There can be AuditV6, AuditV7, AuditV8 ... 
The Model class abstracts that. However for ES 6xx and 7xx we are going to use the same underlying class Audit

If the underlying class / document structure is going to be changed in future say something changes in ES 8 or we add new structure, this becomes handy to abstract that out.

Reply:
I don't think this is necessary. When needed the class hierarchy can be changed, especially that 6.x model should be the same as 7.x model and it is not clear whether 8.x model will be different.

Reply:
I agree it is not necessary at the moment but implemented it for consistency with other security configs and also to make it easier if the doc structure were to change. Even the recent NodeDn reload was done in a similar way https://github.com/krishnaggk/security/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/DynamicConfigFactory.java#L330 

Let me know if you would like me to remove it.

Reply:
Yes, I'd like to remove it. Please see https://github.com/vrozov/security/tree/auditlog